### ISSUE TITLE
🚀 Feature: Add `hasNextPage` indicator to a `DocumentList` object.

### ISSUE THREAD

**Issue by p4-k4** [appwrite/appwrite#3539]
### 🔖 Feature description

Add `hasNextPage` indicator to a `DocumentList` object indicating whether we've reached the start or end of a given query. 

### 🎤 Pitch

Allows for simpler pagination implementation and convenient access.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by Meldiron** [comment]
TLDR; By fetching one extra document purely to do this check, you can know if you have the next page.

We had this discussion many times. No matter how much we tried, there wasn't a way to do proper `hasNextPage` and `hasPreviousPage` without having to do 2 SQL queries. That we surely don't want to do for each request, as it will slow everything down even if you don't need it.

You can solve this on your side for now easily (considering your page is 25 items):
1. ListDocuments, set limit to 26
2. If you get 26 documents as response, mark `hasNextPage = true`. Else, `hasNextPage = false`
3. Delete 26th document from the response, if there is one
4. Use 25 items in the rest of your code

Would this trick solve your problem?

**Comment by p4-k4** [comment]
@Meldiron  Sorry about that. I had posted this FR as it seemed that it was still worthwhile making judging from what you mentioned.
<img width="767" alt="Screen Shot 2022-07-19 at 12 09 31 AM" src="https://user-images.githubusercontent.com/61640764/179508412-80b642d3-1cac-4e91-8acf-abc2807b42c9.png">

I've found a couple of different ways to take care of this (as mentioned on Discord). When using your suggestion I still had to implement some clunky logic in order to check start/end; specifically when changing cursor direction halfway through, or starting from a cursor position that is somewhere in the middle of the collection and navigating back to a start or end from there.

In the end, I'm just catching exceptions where a response has a document count less than the page size or is empty. I'm thinking a solution could be built into the SDK and we won't even need to rely on the server thus jeopardising performance.

There's probably a more elegant solution but I'm no expert and maybe others might still be looking for a simpler solution.

**Comment by pomarec** [comment]
It seems to me that `result.total` gives you the correct count of results for your query

**Comment by eldadfux** [comment]
Stale issue, closing down
